src/ered.erl

```diff
@@ -1,174 +1,196 @@
+    ered_cluster:command(Pid, Command, Key, infinity);
+command({client, Pid}, Command, Key) when is_binary(Key)  ->
+    ered_client:command(Pid, Command, infinity);
+command({client, Pid}, Command, Timeout) when is_integer(Timeout); Timeout =:= infinity ->
+    ered_client:command(Pid, Command, Timeout).
+command({cluster, Pid}, Command, Key, Timeout) ->
+    ered_cluster:command(Pid, Command, Key, Timeout);
+command({client, Pid}, Command, _Key, Timeout) ->
+    ered_client:command(Pid, Command, Timeout).
+
+%% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+-spec command_async(ered_ref(), command(), fun((reply()) -> any())) -> ok.
+-spec command_async(ered_ref(), command(), key(), fun((reply()) -> any())) -> ok.
 %%
-%% Like command/3,4 but asynchronous. Instead of returning the reply, the reply
-%% function is applied to the reply when it is available. The reply function
-%% runs in an unspecified process.
-%% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-command_async(ServerRef, Command, Key, ReplyFun) when is_function(ReplyFun, 1) ->
-    ered_cluster:command_async(ServerRef, Command, Key, ReplyFun).
-
-%% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
--spec command_all(server_ref(), command()) -> [reply()].
--spec command_all(server_ref(), command(), timeout()) -> [reply()].
+%% Like command/2,3,4 but asynchronous. Instead of returning the reply,
+%% the reply function is applied to the reply when it is available.
+%% The reply function runs in an unspecified process and should not
+%% hang or perform any lengthy task.
+%% For cluster clients, a key must be provided.
+%% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+command_async({client, Pid}, Command, ReplyFun) when is_function(ReplyFun, 1) ->
+    ered_client:command_async(Pid, Command, ReplyFun).
+command_async({cluster, Pid}, Command, Key, ReplyFun) when is_function(ReplyFun, 1) ->
+    ered_cluster:command_async(Pid, Command, Key, ReplyFun);
+command_async({client, Pid}, Command, _Key, ReplyFun) when is_function(ReplyFun, 1) ->
+    ered_client:command_async(Pid, Command, ReplyFun).
+
+%% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+-spec command_all(cluster_ref(), command()) -> [reply()].
+-spec command_all(cluster_ref(), command(), timeout()) -> [reply()].
 %%
-%% Send the same command to all connected master Redis nodes.
+%% Send the same command to all primary nodes in the cluster.
 %% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-command_all(ServerRef, Command) ->
-    ered_cluster:command_all(ServerRef, Command, infinity).
-
-command_all(ServerRef, Command, Timeout) ->
-    ered_cluster:command_all(ServerRef, Command, Timeout).
+command_all({cluster, Pid}, Command) ->
+    ered_cluster:command_all(Pid, Command, infinity).
+command_all({cluster, Pid}, Command, Timeout) ->
+    [ered_client:command(Pid, Command, Timeout)].
```

Comment:
Shouldn't it be `ered_cluster:command_all` here?

Reply:
Oops! Should maybe add some tests for the standalone API....